spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -486,6 +488,11 @@ else if (numCompletedFutures.incrementAndGet() == numExpectedFutures) {
 		return settableListenableFuture;
 	}
 
+	private SubscriberStub getSubscriberStub(String subscription) {
+		subscriptionNameToStubMap.putIfAbsent(subscription, this.subscriberFactory.createSubscriberStub(subscription));
```

Comment:
Similar to the `putIfAbsent` comment in autoconfig -- don't create an object (` this.subscriberFactory.createSubscriberStub(subscription)`), unless you already know it does not exist. Otherwise we'd create objects only to throw them away. 

So, "if stub already in the map, just return that. If not, then return a new one from subscriber factory.